var/spack/repos/builtin/packages/libxc/package.py

```diff
@@ -36,6 +36,10 @@ class Libxc(Package):
     version('2.2.2', 'd9f90a0d6e36df6c1312b6422280f2ec')
     version('2.2.1', '38dc3a067524baf4f8521d5bb1cd0b8f')
 
+    # libxc provides a testsuite, but many tests fail
+    # http://www.tddft.org/pipermail/libxc/2013-February/000032.html
+    phases = ['configure', 'build', 'install']
```

Comment:
I don't really like this for 2 reasons:

1. If you don't want to run the unit tests, don't use `spack install --test=all`. If you do want to run them (even though they are expected to fail) you should still be able to.
2. This is a really hacky way of disabling testing. At first glance it looks like you're removing `autoreconf` from the list of phases, so it isn't clear why it disables testing. It turns out that redefining the list of phases also unsets all `run_before` and `run_after` steps like unit testing. I would consider this to be a bug. We shouldn't use bugs as if they were features. A more explicit way of disabling testing would be:
```python
def check(self):
    # libxc provides a testsuite, but many tests fail
    # http://www.tddft.org/pipermail/libxc/2013-February/000032.html
    pass
```

Reply:
Nice! Done.

I  concept of skipping the tests was not mine, I just did the recasting into ``AutotoolsPackage`` seeing it would much compactify and standardize the code.